pandas/core/indexing.py

```diff
@@ -2371,12 +2371,15 @@ def ravel(i):
 
             # we have a frame, with multiple indexers on both axes; and a
             # series, so need to broadcast (see GH5206)
-            if sum_aligners == self.ndim and all(is_sequence(_) for _ in indexer):
+            if all(is_sequence(_) or isinstance(_, slice) for _ in indexer):
                 ser_values = ser.reindex(obj.axes[0][indexer[0]])._values
 
                 # single indexer
                 if len(indexer) > 1 and not multiindex_indexer:
-                    len_indexer = len(indexer[1])
+                    if isinstance(indexer[1], slice):
+                        len_indexer = len(obj.loc[indexer[1]].axes[1])
```

Comment:
I think you can use `axes[1][indexer[1]]` instead of `.loc`.

Reply:
I've made changes by suggestion, but failed test appeared